keps/core/model-cross-worksapce-entitlements.md

```diff
@@ -0,0 +1,1004 @@
+that use this API to implement restrictions to the use of their concepts.
+
+Additionally, a default implemenation will be provided for `kcp start` to allow for
+boot-strappiung the system. This implemenation will be self-sufficient at small scales,
+but the expectation will be that users with more complex entitlement models will use
+third-party systems based on more performant databases to implement their entitlement
+reviews. 
+
+### User Stories
+
+#### System Administrators Can Configure External Entitlement Checks
+
+A system administrator is using `kcp` to provide a multi-tenant platform for developers
+at Acme Corp. Internally, Acme Corp. records which developers or teams have access to
+which services. By implementing a delegate entitlement review webhook, the system
+administrator or platform team can connect `kcp` entitlement checks to the external
+system of record.
+
+#### Service Providers On `kcp` Can Gate Access To Their Service
+
+A team at Acme Corp invents a new cross-workspace paradigm whereby consumers enable
+their workspaces to make use of the paradigm. In order to gate access to this new
+functionality, the provider team writes a validating admission webhook and uses the
+standard `kcp` entitlement review API to ensure only authorized use of their paradigm
+is allowed.
+
+### Notes/Constraints/Caveats
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+A new endpoint will be exposed to facilitate entitlement review. Users will `POST`
+an `EntitlementReview` object to the endpoint and synchronously receive the result
+in the response. The `EntitlementReview` has the following form:
+
+```go
+type EntitlementReview struct {
+    metav1.TypeMeta `json:",inline"`
+    // Spec holds information about the request being evaluated.
+    Spec EntitlementReviewSpec `json:"spec"`
+
+    // Status is filled in by the server and indicates whether the request is allowed or not.
+    // +optional
+    Status EntitlementReviewStatus `json:"status,omitempty"`
+}
+
+// EntitlementReviewSpec is a description of the entitlement request.
+type EntitlementReviewSpec struct {
+    // UserInfo is information about the requesting user
+    UserInfo authenticationv1.UserInfo `json:"userInfo"`
+
+    // RequestInfo is information about the context in which this request was issued.
+    RequestInfo RequestInfo `json:"requestInfo"`
```

Comment:
aren't these implicit in the request handling of the apiserver? No need to have them in the object posted by the client.